core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -241,6 +241,26 @@ private List<TestTemplateInvocationContext> processClusterTestInternal(Extension
                 .collect(Collectors.toList());
     }
 
+    private List<TestTemplateInvocationContext> repeatTestContexts(
+        List<TestTemplateInvocationContext> contexts
+    ) {
+        int count;
+        try {
+            String repeatCount = System.getProperty("kafka.cluster.test.repeat", "1");
+            count = Integer.parseInt(repeatCount);
+        } catch (NumberFormatException e) {
+            count = 1;
+        }
+        if (count <= 1) {
+            return contexts;
+        }
+        List<TestTemplateInvocationContext> repeatedContexts = new ArrayList<>(contexts.size() * count);
+        for (int i = 0; i < count; i++) {
+            repeatedContexts.addAll(contexts);
```

Comment:
We don't create copy of contexts, so those contexts must a kind of immutable objects to avoid corruption caused by other run. Hence, could you add comments to `ZkClusterInvocationContext` and `RaftClusterInvocationContext` to highlight the requisite. Also, could you please move the `clusterReference` of `ZkClusterInvocationContext` to be a local variable to make `ZkClusterInvocationContext` be immutable.

https://github.com/apache/kafka/blob/trunk/core/src/test/java/kafka/test/junit/ZkClusterInvocationContext.java#L78

Reply:
This is a good point. I'll modify the annotation processing to create unique instances of the invocation contexts.